### PR TITLE
add quazip recipe

### PR DESCRIPTION
**Pull request by izahn**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by izahn** [comment]
Ready for review @conda-forge/help-c-cpp

File: recipes/quazip/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir build
+
+cmake -B build -S . -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib
```
Comment (wolfv) [inline recipes/quazip/build.sh:5]: maybe you could use `${CMAKE_ARGS}`? That would also help cross-compiling in the future. 
You can also do that for the Windows build file.

File: recipes/quazip/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% set name = "quazip" %}
+{% set version = "1.1" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/stachenov/quazip/archive/refs/tags/v1.1.tar.gz
```
Comment (wolfv) [inline recipes/quazip/meta.yaml:9]: Not using `{{ version }}` in the URL here. Would be better for the autotick bot to do its magic.

**Comment by wolfv** [comment]
Looks good, left two comments for improvements.

File: recipes/quazip/meta.yaml (hunk 2 above)
Comment (izahn) [reply recipes/quazip/meta.yaml]: Done in https://github.com/conda-forge/staged-recipes/pull/16073/commits/ea0f2dccd3b05066ceba40ae935ffbd3c85f29a2 , thanks!

**Comment by izahn** [comment]
Thank you @wolfv  , I used CMAKE_ARGS and the version template as you suggested.

**Pull request merged**